File: go/space-age/space_age.go
package space

// Planet type is just a string
type Planet = string

// Age returns age on planet
func Age(time float64, planet Planet) float64 {
	table := map[Planet]float64{
		"Earth":   1.,
		"Mercury": 0.2408467,
		"Venus":   0.61519726,
		"Mars":    1.8808158,
		"Jupiter": 11.862615,
		"Saturn":  29.447498,
		"Uranus":  84.016846,
		"Neptune": 164.79132,
	}

	return time / (table[planet] * 31557600)
}

File: haskell/space-age/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

ageOn :: Planet -> Float -> Float
ageOn planet seconds =
  let earthSeconds = 365.25 * 24 * 60 * 60
  in case planet of
    Mercury   -> seconds / (0.2408467 * earthSeconds)
    Venus     -> seconds / (0.61519726 * earthSeconds)
    Earth     -> seconds / (1.0 * earthSeconds)
    Mars      -> seconds / (1.8808158 * earthSeconds)
    Jupiter   -> seconds / (11.862615 * earthSeconds)
    Saturn    -> seconds / (29.447498 * earthSeconds)
    Uranus    -> seconds / (84.016846 * earthSeconds)
    Neptune   -> seconds / (164.79132 * earthSeconds)

